packages/editor/src/editor/plugins/__tests__/createWithInsertData.test.tsx

```diff
@@ -0,0 +1,153 @@
+import {describe, expect, it, jest} from '@jest/globals'
+import {type Descendant} from 'slate'
+
+import {
+  type PortableTextMemberSchemaTypes,
+  type PortableTextSlateEditor,
+} from '../../../types/editor'
+import {exportedForTesting} from '../createWithInsertData'
+
+const initialValue = [
+  {
+    _key: 'a',
+    _type: 'myTestBlockType',
+    children: [
+      {
+        _key: 'a1',
+        _type: 'span',
+        marks: ['link1'],
+        text: 'Block A',
+      },
+      {
+        _key: 'a2',
+        _type: 'span',
+        marks: ['colour1'],
+        text: 'Block B',
+      },
+    ],
+    markDefs: [
+      {
+        _key: 'link1',
+        _type: 'link',
+        href: 'google.com',
+        newTab: false,
+      },
+      {
+        _key: 'colour1',
+        _type: 'color',
+        color: 'red',
+      },
+    ],
+    style: 'normal',
+  },
+]
+
+describe('plugin: createWithInsertData _regenerateKeys', () => {
+  it('has MarkDefs that are allowed annotations', async () => {
+    const {_regenerateKeys} = exportedForTesting
+    const editorTypes = {annotations: [{name: 'color'}, {name: 'link'}]}
+    const editor = {
+      isTextBlock: jest.fn().mockReturnValue(true),
+      isTextSpan: jest.fn().mockReturnValue(true),
+    } as unknown as PortableTextSlateEditor
+
+    const generatedValue = _regenerateKeys(
+      editor,
+      initialValue as Descendant[],
+      jest.fn().mockReturnValue('') as () => string,
+      'span',
+      editorTypes as unknown as PortableTextMemberSchemaTypes,
+    )
+
+    // the keys are not important here as it's not what we are testing here
+    expect(generatedValue).toStrictEqual([
+      {
+        _key: '',
+        _type: 'myTestBlockType',
+        children: [
+          {_key: '', _type: 'span', marks: [''], text: 'Block A'},
+          {
+            _key: '',
+            _type: 'span',
+            marks: [''],
```

Comment:
Just so I understand, is this actually what we expect? Why is `marks: ['']` in this test case? Shouldn't we preserve all marks in this teset?

Reply:
Ah this is because of my mock for the generateKey 
     ` jest.fn().mockReturnValue('') as () => string,`

Because I wasn't concerned about what key it was, I just wanted to make sure it was spitting the right number of marks and markDefs, the value of the keys themselves don't really matter

That being said, probably best to do something like 

     ` jest.fn().mockReturnValue('dummy') as () => string,`

on line 57

Reply:
Mentioned it in the comment at line 62 :)